### PR TITLE
Fixes Camera.project_position to not return get_global_transform().origin if projection mode is orthogonal

### LINKED ISSUE #33837: Camera.project_position return camera's globla_transform.origin when the camera'projection is orthogonal and the parameter depth is 0,
**Godot version:**
Godot 3.2 beta 1

**OS/device including version:**
Windows 10 64 bits

**Issue description:**
When the camera's projection is orthogonal and the parameter depth is 0, project_position will return the camera's globla_transform.origin, yes, if the projection is perspective or frustum, this is of course correct.

**Steps to reproduce:**
Create new 3D scene, put a camera, change projection to orthogonal  andwrite print(project_position(get_viewport().get_mouse_position(),0))
print(project_position(get_viewport().get_mouse_position(),1)) 

**Minimal reproduction project:**

### PR DESCRIPTION
**Pull request by fsy98**
Fix #33837

### CODE REVIEW

File: scene/3d/camera.cpp [hunk 1]
```diff
@@ -390,10 +390,9 @@ Vector3 Camera::project_position(const Point2 &p_point, float p_z_depth) const {
 
 	ERR_FAIL_COND_V_MSG(!is_inside_tree(), Vector3(), "Camera is not inside scene.");
 
-	if (p_z_depth == 0) {
+	if (p_z_depth == 0 && mode!= PROJECTION_ORTHOGONAL) {
```
Comment (akien-mga) [inline scene/3d/camera.cpp]: There's a style issue, there should be spaces on both sides of `!=`.

**Comment by akien-mga** [comment]
Could you squash commits together, and give the resulting commit a more explicit commit message? See https://github.com/godotengine/godot/blob/master/CONTRIBUTING.md#format-your-commit-logs-with-readability-in-mind and [PR workflow](http://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#mastering-the-pr-workflow-the-rebase).

**Comment by fsy98** [comment]
is wired that I squash commits together,but it make mistake

**Comment by akien-mga** [comment]
You still have 2 commits, with `Merge pull request #1 from godotengine/master` which should not be included in this PR.

And the commit message for the second commit is:
```
Update camera.cpp

Update camera.cpp
```
which is not explicit at all.

**Comment by fsy98** [comment]
Sorry I am totally new to the GitHub, I focus change the  commit message, but I am confused why after I did the squash commits, it still shows Merge pull request #1 from godotengine/master …

**Comment by fsy98** [comment]
I don't know why I in that commit, is not done by me, and it seams that is the master of my commit, did any one have a good idea to solve that?

**Comment by akien-mga** [comment]
Sorry for the delay.

The first commit is there because one of your team member used a Pull Request to update your fork (https://github.com/Team-Bad/godot/pull/1) instead of doing a manual pull. GitHub sadly lacks an option in its interface to update a branch without custom modifications to the latest upstream master branch.

I fixed it myself by rebasing on the upstream master branch, and fixed the style issues that were remaining (missing space before `!=`, and trailing tabs after `}`). I could not force-update the PR as modifications from maintainers have been disabled, so I pushed the fixed commit directly: d80e979a48.

Thanks for the contribution!

**Pull request closed without merging**